### PR TITLE
Update Ocean to blaze 3.8

### PR DESCRIPTION
**Pull request by geoffrey4444**
## Proposed changes

Update Blaze on ocean to v3.8.

### Upgrade instructions

This PR just switches to use blaze 3.8 on the cluster Ocean at Cal State Fullerton, which is needed after spectre switched to requiring blaze 3.8.

### Code review checklist

- [x] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [x] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [x] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

**Comment by fmahebert** [comment]
[not a change request] What about the other Ocean environments? Are they still supported?

**Comment by kidder** [comment]
@geoffrey4444 this needs a rebase

**Review by moxcodes — DISMISSED** [review]

**Comment by geoffrey4444** [comment]
Thanks for taking the quick looks, @kidder and @fmahebert ! I rebased on latest develop and made the following changes:

1) I continue to support ocean_clang.sh and ocean_gcc.sh. Both use the same modules but the clang one uses clang instead of gcc when compiling spectre. Both of these are suitable for running jobs on ocean's orca-1 queue (the newer, faster queue).
2) I dropped support for ocean_orca0_gcc.sh. This let us run on our older compute nodes but required compiling on those older, slower compute nodes, which was a big pain. Once we're doing a lot of production runs with spectre, I might bring this support back with updated modules, but until then, I decided it's not worth the time, since no one was actually running spectre on the orca0 queue as long as I can remember.

File: support/Environments/ocean_gcc.sh [hunk 1]
```diff
@@ -18,24 +18,25 @@ spectre_setup_modules() {
...
-    module unload libxsmm-1.10-gcc-7.3.0-sjh5yzv
+    module unload libxsmm/1.16.1
     module unload yaml-cpp-develop-gcc-7.3.0-qcfbbll
     module unload boost-1.68.0-gcc-7.3.0-vgl6ofr
-    module unload hdf5-1.10.4-gcc-7.3.0-ytt4j54
+    module unload hdf5-1.12.0-gcc-7.3.0-mknp6xv
     module unload openblas-0.3.4-gcc-7.3.0-tt2coe7
     module unload python/3.7.0
-    module unload charm-6.10.2-libs
+    module unload charm-6.10.2-libss
```
Comment (kidder) [inline support/Environments/ocean_gcc.sh]: fix this

**Review by kidder — CHANGES_REQUESTED** [review]
you can squash

**Review by fmahebert — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**